apps/condo/domains/miniapp/access/B2BAppContext.js

```diff
@@ -9,44 +9,54 @@ const { getById } = require('@open-condo/keystone/schema')
 
 const { checkB2BAppAccessRight } = require('@condo/domains/miniapp/utils/accessSchema')
 const { checkOrganizationPermission } = require('@condo/domains/organization/utils/accessSchema')
-const { SERVICE } = require('@condo/domains/user/constants/common')
+const { SERVICE, STAFF } = require('@condo/domains/user/constants/common')
+const { canDirectlyReadSchemaObjects, canDirectlyManageSchemaObjects } = require('@condo/domains/user/utils/directAccess')
 
 
 /**
  * B2B App context may only be read:
  * 1. Admin / support
- * 2. Organization employee (without settings / state)
- * 3. Integration service account
+ * 2. Users with direct access
+ * 3. Organization employee (without settings / state)
+ * 4. Integration service account
  */
-async function canReadB2BAppContexts ({ authentication: { item: user } }) {
+async function canReadB2BAppContexts ({ authentication: { item: user }, listKey }) {
     if (!user) return throwAuthenticationError()
     if (user.deletedAt) return false
 
     if (user.isAdmin || user.isSupport) return {}
 
+    const hasDirectAccess = await canDirectlyReadSchemaObjects(user, listKey)
+    if (hasDirectAccess) return true
```

Comment:
```suggestion
    if (typeof hasDirectAccess !== "undefined") return hasDirectAccess
```

Reply:
Why? Its returns only boolean, since you can read / manage anything or nothing